### DISCUSSION TITLE
nekrs sample test fails with CUDA_ERROR_INVALID_DEVICE

### DISCUSSION THREAD

**Discussion by nandini1610** [Nek5000/nekRS#438] [General]
Running the sample test mpirun -np 2 nekrs --setup turbPipe.par resulted in the following error running on a GPU server.
Initializing device
terminate called after throwing an instance of 'occa::exception'
  what():
---[ Error ]--------------------------------------------------------------------
    File     : /program/nekrs-22.0/nekRS-22.0/3rd_party/occa/src/occa/internal/modes/cuda/device.cpp
    Line     : 30
    Function : device
    Message  : Device: Creating Device
               CUDA Error [ 101 ]: CUDA_ERROR_INVALID_DEVICE
    Stack
      12 occa::error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
      11 occa::cuda::error(cudaError_enum, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
      10 occa::cuda::device::device(occa::json const&)
       9 occa::cuda::cudaMode::newDevice(occa::json const&)
       8 occa::device::setup(occa::json const&)
       7 occa::device::setup(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
       6 device_t::device_t(setupAide&, comm_t&)
       5 platform_t::platform_t(setupAide&, int, int)
       4 nekrs::setup(int, int, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int)
       3 nekrs()
       2 /lib64/libc.so.6(__libc_start_main+0xf5)
       1 nekrs()
====================================================

**Comment by yslan** [comment]
Does you machine has two GPU?
What does nvidia-smi shows?

**Reply by nandini1610** [reply]
this is what it shows?
nvidia-smi
Fri Jun 17 13:41:00 2022
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.82.01    Driver Version: 470.82.01    CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-PCIE...  On   | 00000001:00:00.0 Off |                    0 |
| N/A   25C    P0    25W / 250W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

**Reply by nandini1610** [reply]
The last run was done on a node with 1GPU. This time I ran on a node with 2 GPU and the cuda error is gone but still i have this error.
 mpirun -np 2 nekrs --setup turbPipe.par
                 __    ____  _____
   ____   ___   / /__ / __ \/ ___/
  / __ \ / _ \ / //_// /_/ /\__ \
 / / / //  __// ,<  / _, _/___/ /
/_/ /_/ \___//_/|_|/_/ |_|/____/  v22.0.0 (no sha)

COPYRIGHT (c) 2019-2022 UCHICAGO ARGONNE, LLC

MPI tasks: 2

reading par file ...

using NEKRS_HOME: /program/nekrs-22.0/nekrs
using NEKRS_CACHE_DIR: /enc/udev_ijnLT/work/shared/turbPipePeriodic/.cache
using OCCA_CACHE_DIR: /enc/udev_ijnLT/work/shared/turbPipePeriodic/.cache/occa/

Initializing device
active occa mode: CUDA

terminate called after throwing an instance of 'occa::exception'
  what():
---[ Error ]--------------------------------------------------------------------
    File     : /program/nekrs-22.0/nekRS-22.0/3rd_party/occa/src/occa/internal/modes/cuda/device.cpp
    Line     : 351
    Function : compileKernel
    Message  : Error compiling [myDummyKernelName], Command: [nvcc -O3 --fmad=true -lineinfo -arch=sm_70 -fatbin -Xptxas -v -I/program/nekrs-22.0/nekrs/include -I/program/nekrs-22.0/nekrs/include -L/program/nekrs-22.0/nekrs/lib -locca -x cu /enc/udev_ijnLT/work/shared/turbPipePeriodic/.cache/occa/cache/8883875db6215ce2/source.cpp -o /enc/udev_ijnLT/work/shared/turbPipePeriodic/.cache/occa/cache/8883875db6215ce2/binary > /dev/null 2>&1]
    Stack
      12 occa::error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
      11 occa::cuda::device::compileKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&, occa::io::lock_t&)
      10 occa::cuda::device::buildKernelFromProcessedSource(occa::hash_t, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, occa::lang::sourceMetadata_t&, occa::lang::sourceMetadata_t&, occa::json const&, occa::io::lock_t)
       9 occa::launchedModeDevice_t::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::hash_t, bool, occa::json const&)
       8 occa::launchedModeDevice_t::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::hash_t, occa::json const&)
       7 occa::device::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&) const
       6 occa::device::buildKernelFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&) const
       5 platform_t::platform_t(setupAide&, int, int)
       4 nekrs::setup(int, int, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int)
       3 nekrs()
       2 /lib64/libc.so.6(__libc_start_main+0xf5)
       1 nekrs()
================================================================================

**Reply by nandini1610** [reply]
I also upgraded the cuda version and tested and i get this error.
 mpirun -np 2 nekrs --setup turbPipe.par 

MPI tasks: 2

reading par file ...

using NEKRS_HOME: /program/nekrs-22.0/nekrs
using NEKRS_CACHE_DIR: /enc/udev_DunLT/work/shared/turbPipePeriodic/.cache
using OCCA_CACHE_DIR: /enc/udev_DunLT/work/shared/turbPipePeriodic/.cache/occa/

Initializing device 
[[CUDACUDA] mode is not enabled, defaulting to [Serial] mode
] mode is not enabled, defaulting to [Serial] mode
active occa mode: Serial

active occa mode does not match selected backend!
Abort(1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
Abort(1) on node 1 (rank 1 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 1

**Comment by yslan** [comment]
I assume you can run single GPU without issue?

- From your `nvidia-smi` output, I can only find 1 GPU.
  First, I'd make sure you do have 2 visible GPU on your "2 gpu-node". 
  `nvidia-smi` should show multiple GPU id and their status

- The error `Error compiling [myDummyKernelName], ` can be related to many things.
  I'd recommend to remove the local cache before next actions whenever you found an error. 
  ```
    (in your case, it's at)
    /enc/udev_DunLT/work/shared/turbPipePeriodic/.cache
  ```

- As for the final error
  ```
  [[CUDACUDA] mode is not enabled, defaulting to [Serial] mode
  ```
  It looks like the OCCA cannot run with cuda successfully.
  My guess is, you didn't rebuild the course code after upgrading the cuda version?
  ```
  goto <NEKRS_HOME> 
  rm -r build
  ./nrsconfig
  ...

**Reply by nandini1610** [reply]
I recompiled and I dont see the [Serial] mode error.  And I am now testing on a single GPU node using mpirun -np 1 and still I see the Error compiling and I have also deleted the .cache dir before running. Though i delete i see it is recreated by the nekrs process.
MPI tasks: 1

reading par file ...

using NEKRS_HOME: /program/nekrs-22.0/nekrs/
using NEKRS_CACHE_DIR: /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache
using OCCA_CACHE_DIR: /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/

Initializing device
active occa mode: CUDA

terminate called after throwing an instance of 'occa::exception'
  what():
---[ Error ]--------------------------------------------------------------------
    File     : /program/nekrs-22.0/nekRS-22.0/3rd_party/occa/src/occa/internal/modes/cuda/device.cpp
    Line     : 351
    Function : compileKernel
    Message  : Error compiling [myDummyKernelName], Command: [nvcc -O3 --fmad=true -lineinfo -arch=sm_37 -fatbin -Xptxas -v -I/program/nekrs-22.0/nekrs//include -I/program/nekrs-22.0/nekrs//include -L/program/nekrs-22.0/nekrs//lib -locca -x cu /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/cache/739520af4546e231/source.cpp -o /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/cache/739520af4546e231/binary > /dev/null 2>&1]
    Stack
      12 occa::error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
      11 occa::cuda::device::compileKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&, occa::io::lock_t&)
      10 occa::cuda::device::buildKernelFromProcessedSource(occa::hash_t, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, occa::lang::sourceMetadata_t&, occa::lang::sourceMetadata_t&, occa::json const&, occa::io::lock_t)
       9 occa::launchedModeDevice_t::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::hash_t, bool, occa::json const&)
       8 occa::launchedModeDevice_t::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::hash_t, occa::json const&)
       7 occa::device::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&) const
       6 occa::device::buildKernelFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&) const
       5 platform_t::platform_t(setupAide&, int, int)
       4 nekrs::setup(int, int, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int)
       3 nekrs()
       2 /lib64/libc.so.6(__libc_start_main+0xf5)
       1 nekrs()
================================================================================


===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 29616 RUNNING AT ip-10-23-20-201.ec2.internal
=   KILLED BY SIGNAL: 6 (Aborted)

**Reply by yslan** [reply]
What happens if you execute the following command causing error (keep the .cache this time):
```
nvcc -O3 --fmad=true -lineinfo -arch=sm_37 -fatbin -Xptxas -v -I/program/nekrs-22.0/nekrs//include -I/program/nekrs-22.0/nekrs//include -L/program/nekrs-22.0/nekrs//lib -locca -x cu /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/cache/739520af4546e231/source.cpp -o /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/cache/739520af4546e231/binary
```

Yes, the `.cache` is where we put the compiled kernels from just-in-time compilation.
remove that will force a "make clean" to your case.

**Reply by nandini1610** [reply]
I am seeing this serial mode message again. I do the compilation in the same way as I did the previous install except this one uses gcc 6.5.0 and previous one 8.3.0.

mpirun -np 1 nekrs --setup turbPipe.par
                 __    ____  _____
   ____   ___   / /__ / __ \/ ___/
  / __ \ / _ \ / //_// /_/ /\__ \
 / / / //  __// ,<  / _, _/___/ /
/_/ /_/ \___//_/|_|/_/ |_|/____/  v22.0.0 (no sha)

COPYRIGHT (c) 2019-2022 UCHICAGO ARGONNE, LLC

MPI tasks: 1

reading par file ...

using NEKRS_HOME: /program/nekrs-22.0/nekrs/
using NEKRS_CACHE_DIR: /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache
using OCCA_CACHE_DIR: /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/

Initializing device
[CUDA] mode is not enabled, defaulting to [Serial] mode
active occa mode: Serial

active occa mode does not match selected backend!
Abort(1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[root@ip-10-23-72-83 /program/nekrs-22.0/nekrs/examples/turbPipePeriodic]$ ldd nekrs
ldd: ./nekrs: No such file or directory
[root@ip-10-23-72-83 /program/nekrs-22.0/nekrs/examples/turbPipePeriodic]$ which nekrs
/program/nekrs-22.0/nekrs/bin/nekrs
[root@ip-10-23-72-83 /program/nekrs-22.0/nekrs/examples/turbPipePeriodic]$ ldd $(which nekrs)
	linux-vdso.so.1 =>  (0x00007ffde41cf000)
	libnekrs.so => /program/nekrs-22.0/nekrs/lib/libnekrs.so (0x00002b5f8a475000)
	libocca.so => /program/nekrs-22.0/nekrs/lib/libocca.so (0x00002b5f8a969000)
	libcudart.so.11.0 => /program/nekrs-22.0/cuda-11.4-8.2.4.15/lib64/libcudart.so.11.0 (0x00002b5f8aecc000)
	libcuda.so.1 => /program/nekrs-22.0/cuda-11.4-8.2.4.15/lib64/libcuda.so.1 (0x00002b5f8b16e000)
	libgomp.so.1 => /program/nekrs-22.0/gcc/6.5.0/lib64/libgomp.so.1 (0x00002b5f8b37c000)
	libmpicxx.so.12 => /opt/rescale/oneapi/2021.4.0/mpi/2021.4.0//lib/libmpicxx.so.12 (0x00002b5f8b5a9000)
	libmpifort.so.12 => /opt/rescale/oneapi/2021.4.0/mpi/2021.4.0//lib/libmpifort.so.12 (0x00002b5f8b7c9000)
	libmpi.so.12 => /opt/rescale/oneapi/2021.4.0/mpi/2021.4.0//lib/release/libmpi.so.12 (0x00002b5f8bb7d000)
	librt.so.1 => /lib64/librt.so.1 (0x00002b5f8d24d000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00002b5f8d455000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00002b5f8d671000)
	libstdc++.so.6 => /program/nekrs-22.0/gcc/6.5.0/lib64/libstdc++.so.6 (0x00002b5f8d875000)
	libm.so.6 => /lib64/libm.so.6 (0x00002b5f8dbf7000)
	libgcc_s.so.1 => /program/nekrs-22.0/gcc/6.5.0/lib64/libgcc_s.so.1 (0x00002b5f8def9000)
	libc.so.6 => /lib64/libc.so.6 (0x00002b5f8e10f000)
	libgfortran.so.3 => /program/nekrs-22.0/gcc/6.5.0/lib64/libgfortran.so.3 (0x00002b5f8e4dd000)
	libquadmath.so.0 => /program/nekrs-22.0/gcc/6.5.0/lib64/libquadmath.so.0 (0x00002b5f8e803000)
	/lib64/ld-linux-x86-64.so.2 (0x00002b5f8a251000)
[root@ip-10-23-72-83 /program/nekrs-22.0/nekrs/examples/turbPipePeriodic]$ nvcc -O3 --fmad=true -lineinfo -arch=sm_37 -fatbin -Xptxas -v -^C
[root@ip-10-23-72-83 /program/nekrs-22.0/nekrs/examples/turbPipePeriodic]$ nvcc -O3 --fmad=true -lineinfo -arch=sm_37 -fatbin -Xptxas -v -I/program/nekrs-22.0/nekrs//include -I/program/nekrs-22.0/nekrs//include -L/program/nekrs-22.0/nekrs//lib -locca -x cu /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/cache/739520af4546e231/source.cpp -o /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/cache/739520af4546e231/binary
nvcc warning : The 'compute_35', 'compute_37', 'compute_50', 'sm_35', 'sm_37' and 'sm_50' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
gcc: error: /program/nekrs-22.0/nekrs/examples/turbPipePeriodic/.cache/occa/cache/739520af4546e231/source.cpp: No such file or directory
gcc: warning: ‘-x c++’ after last input file has no effect
gcc: fatal error: no input files
compilation terminated.

**Reply by yslan** [reply]
Same thing here. After changing the compiler, the installed nekrs source including occa will no longer usable. 
As long as the occa can't work properly, you will always fall back to SERIAL mode and see the error "`active occa mode does not match selected backend!`"

The rebuild is needed after each time the compiler changes. (`cd <nekrs source folder>; rm -r build; ./nrsconfig; ....; cd <case folder>;rm -r .cache;`).
And, I will expect you still get the same error (`Error compiling [myDummyKernelName]`)  when you try to run the example with single GPU
Now, it will show up the command it stucks.
You just need to copy and execute the command causing error without redirect the stderr to background `> /dev/null 2>&1`.
So, you can get the true error why the compilation fails.

**Reply by nandini1610** [reply]
Everytime I do the instal, it  is a fresh installation and it is not reusing any existing one.If you look at the nekrs binaries it shows it has been compiled properly. So I tried recompiling multiple times with different cuda version on gcc 8.3.0, I get the  same error  
"active occa mode does not match selected backend!"

**Reply by nandini1610** [reply]
Though i have compiled using the cuda and this is the o/p when i check whether occa uses cuda or not.
cd nekrs/3rd_party/occa
make -j
 occa info
    ========+======================+===========================================
     CPU(s) | Processor Name       | Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
            | Memory               | 59.96 GB
            | Clock Frequency      |
            | SIMD Instruction Set | AVX2
            | SIMD Width           | 256 bits
            | L1d Cache Size       |  32 KB
            | L1i Cache Size       |  32 KB
            | L2 Cache Size        | 256 KB
            | L3 Cache Size        |  45 MB
    ========+======================+===========================================
     OpenCL | Device Name          | Tesla K80
            | Driver Vendor        | NVIDIA
            | Platform ID          | 0
            | Device ID            | 0
            | Memory               | 11.17 GB
    ========+======================+===========================================

**Reply by yslan** [reply]
This is weird.
Your `nvidia-smi` was showing   0 Tesla V100-PCIE.
However, occa info only picks up the OpenCL backend and it shows `Tesla K80`

**Reply by nandini1610** [reply]
We are on cloud paltform so everytime we get a different set of GPU. So it is all giving the right GPU hardware, please see below.
nvidia-smi
Mon Jun 20 18:50:26 2022
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.82.01    Driver Version: 470.82.01    CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla K80           On   | 00000000:00:1E.0 Off |                    0 |
| N/A   33C    P8    30W / 149W |      0MiB / 11441MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found

> occa info
    ========+======================+===========================================
     CPU(s) | Processor Name       | Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
            | Memory               | 59.96 GB
            | Clock Frequency      |
            | SIMD Instruction Set | AVX2
            | SIMD Width           | 256 bits
            | L1d Cache Size       |  32 KB
            | L1i Cache Size       |  32 KB
            | L2 Cache Size        | 256 KB
            | L3 Cache Size        |  45 MB
    ========+======================+===========================================
     OpenCL | Device Name          | Tesla K80
            | Driver Vendor        | NVIDIA
            | Platform ID          | 0
            | Device ID            | 0
            | Memory               | 11.17 GB

**Comment by nandini1610** [comment]
Finally after active occa mode : CUDA I got the same nvcc error and as you mentioned was able to track down to missing include dir. Let me know what I can do after this step. Should I recompile?

nvcc -O3 --fmad=true -lineinfo -arch=sm_52 -fatbin -Xptxas -v -I/program/nekrs-22.0/nekrs/include -I/program/nekrs-22.0/gcc/8.3.0/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include -I/program/nekrs-22.0/gcc/8.3.0/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include-fixed -L/program/nekrs-22.0/nekrs/lib -locca -x cu /enc/udev_DHqBeb/work/shared/examples/turbPipe/.cache/occa/cache/dc511cf8ce7411de/source.cpp -o /enc/udev_DHqBeb/work/shared/examples/turbPipe/.cache/occa/cache/dc511cf8ce7411de/binary
ptxas info    : 0 bytes gmem
ptxas info    : Compiling entry function '_occa_myDummyKernelName_0' for 'sm_52'
ptxas info    : Function properties for _occa_myDummyKernelName_0
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 2 registers, 324 bytes cmem[0]
